src/storage/store/rels_statistics.cpp

```diff
@@ -32,7 +32,10 @@ void RelsStatistics::setNumRelsForTable(table_id_t relTableID, uint64_t numRels)
     auto relStatistics =
         (RelStatistics*)tablesStatisticsContentForWriteTrx->tableStatisticPerTable[relTableID]
             .get();
-    tablesStatisticsContentForWriteTrx->nextRelID += (numRels - relStatistics->getNumTuples());
+    if (numRels > relStatistics->getNumTuples()) {
```

Comment:
This looks confusing and I actually don't understand the logic. I'd rather be explicit about how  tablesStatisticsContentForWriteTrx->nextRelID is incremented. Note that nextRelID should only be incremented. So I would do it in one of the updateNumRelsByValueForTable if the value is positive. I know setNumRelsForTable is also called by in_mem_rel_copier, so we need a way to set it then too. I can think of 2 things: 

1. Maybe in_mem_rel_copier can also just call one of the updateNumRels functions. 
2. If that won't work, you can have a wrapper function setNumTuples(table_id_t relTableID, uint64_t numRels) which only sets numTuples. And another function incrementNextRelID, which is called both from the updateNumRels function. and in_mem_rel_copier explicitly. 

Let me know if you want to discuss this.

Reply:
I created a new function:
```
void RelsStatistics::updateNumRelsByValue(
    table_id_t relTableID, table_id_t srcTableID, table_id_t dstTableID, int64_t value) {
    lock_t lck{mtx};
    initTableStatisticPerTableForWriteTrxIfNecessary();
    auto relStatistics =
        (RelStatistics*)tablesStatisticsContentForWriteTrx->tableStatisticPerTable[relTableID]
            .get();
    auto numRelsAfterUpdate = relStatistics->getNumTuples() + value;
    relStatistics->setNumTuples(numRelsAfterUpdate);
    for (auto relDirection : REL_DIRECTIONS) {
        auto relStatistics =
            (RelStatistics*)tablesStatisticsContentForWriteTrx->tableStatisticPerTable
                .at(relTableID)
                .get();
        relStatistics->numRelsPerDirectionBoundTable[relDirection].at(
            relDirection == FWD ? srcTableID : dstTableID) += value;
    }
    // Update the nextRelID only when we are inserting rels.
    if (value > 0) {
        tablesStatisticsContentForWriteTrx->nextRelID += value;
    }
    assertNumRelsIsSound(relStatistics->numRelsPerDirectionBoundTable[FWD], numRelsAfterUpdate);
    assertNumRelsIsSound(relStatistics->numRelsPerDirectionBoundTable[BWD], numRelsAfterUpdate);
}
```
It will only update the relID if we are inserting rels.